service/src/main/kotlin/fi/ouka/evakaoulu/invoice/service/EVakaOuluInvoiceClient.kt

```diff
@@ -4,14 +4,44 @@
 
 package fi.ouka.evakaoulu.invoice.service
 
+import com.jcraft.jsch.SftpException
 import fi.espoo.evaka.invoicing.domain.InvoiceDetailed
 import fi.espoo.evaka.invoicing.integration.InvoiceIntegrationClient
-import fi.ouka.evakaoulu.IntimeProperties
+
 
 class EVakaOuluInvoiceClient(
-    private val properties: IntimeProperties
+    private val invoiceSender: InvoiceSender,
+    private val invoiceGenerator: ProEInvoiceGenerator
 ) : InvoiceIntegrationClient {
     override fun send(invoices: List<InvoiceDetailed>): InvoiceIntegrationClient.SendResult {
-        TODO("Not yet implemented")
+        val successList = mutableListOf<InvoiceDetailed>()
+        val failedList = mutableListOf<InvoiceDetailed>()
+        val manuallySentList = mutableListOf<InvoiceDetailed>()
+        var proEinvoices = ""
+
+        val (withSSN, withoutSSN) = invoices.partition { invoice -> invoice.headOfFamily.ssn != null }
+
+        withSSN.forEach {
+            proEinvoices += invoiceGenerator.generateInvoice(it)
+            successList.add(it)
+        }
+
+        try {
+            manuallySentList.addAll(withoutSSN)
+            invoiceSender.send(proEinvoices)
+        } catch (e: SftpException){
+            failedList.addAll(successList)
```

Comment:
Suggest to add logging, at the very least here in error clause

Reply:
Logging added.